frontend/pages/contacts/components/ContactList.tsx

```diff
@@ -0,0 +1,254 @@
+import { ReactComponent as SortIcon } from "@assets/svg/files/sort.svg";
+//
+import { Contact } from "@redux/models/ContactsModels";
+import { CustomInput } from "@components/input";
+import clsx from "clsx";
+import { ChangeEvent, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Fragment } from "react/jsx-runtime";
+import logger from "@/common/utils/logger";
+import { getInitialFromName, removeExtraSpaces } from "@common/utils/strings";
+import { shortAddress } from "@common/utils/icrc";
+import { CustomCopy } from "@components/tooltip";
+import { ChevronDownIcon, ChevronLeftIcon } from "@radix-ui/react-icons";
+import SubAccountTable from "./ICRC/SubAccountTable";
+import ContactAction from "./ICRC/ContactAction";
+import { useAppSelector } from "@redux/Store";
+import { isContactNameValid } from "../helpers/validators";
+import { db } from "@/database/db";
+
+interface ContactListProps {
+  contactSearchKey: string;
+  assetFilter: string[];
+  allowanceOnly: boolean;
+}
+
+enum SortDirection {
+  Asc = "asc",
+  Desc = "desc",
+}
+
+export default function ContactList({ allowanceOnly, assetFilter, contactSearchKey }: ContactListProps) {
+  const { t } = useTranslation();
+  const [contactDropdown, setContactDropdown] = useState<Contact | null>(null);
+  const [contactEdited, setContactEdited] = useState<Contact | null>(null);
+  const [contactNameInvalid, setContactNameInvalid] = useState(false);
+  const [sortDirection, setSortDirection] = useState<SortDirection>(SortDirection.Asc);
+  const contacts = useAppSelector((state) => state.contacts.contacts);
+
+  return (
+    <div className="w-full h-full scroll-y-light max-h-[calc(100vh-12rem)]">
+      <table className="w-full text-PrimaryTextColorLight dark:text-PrimaryTextColor text-md">
+        <thead className="sticky top-0 border-b border-BorderColorTwoLight dark:border-BorderColorTwo text-PrimaryTextColor/70 z-[1]">
+          <tr className="text-PrimaryTextColorLight dark:text-PrimaryTextColor">
+            <th className="p-2 text-left w-[40%] bg-PrimaryColorLight dark:bg-PrimaryColor flex items-center ">
+              <p>{t("contact.name")}</p>
+              <SortIcon
+                className="w-3 h-3 ml-1 cursor-pointer dark:fill-gray-color-6 fill-black-color"
+                onClick={toggleSortDirection}
+              />
+            </th>
+            <th className="p-2 text-left w-[45%] bg-PrimaryColorLight dark:bg-PrimaryColor">
+              <p>{"Principal"}</p>
+            </th>
+            <th className="pr-2 py-2 w-[12%] bg-PrimaryColorLight dark:bg-PrimaryColor">
+              <p>{t("action")}</p>
+            </th>
+            <th className="w-[3%] bg-PrimaryColorLight dark:bg-PrimaryColor"></th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {getSortedContacts(contacts, sortDirection).map((contact, index) => {
+            const isContactExpanded = contactDropdown?.principal === contact.principal;
```

Comment:
This list cannot be send here already applied all those filters? because on each render all this calculations trigger again and again

Reply:
maybe a memo or something that only updates if there is a change. on the values used.